database.cc

```diff
@@ -427,7 +427,7 @@ void backlog_controller::update_controller(float shares) {
     if (!_inflight_update.available()) {
         return; // next timer will fix it
     }
-    _inflight_update = engine().update_shares_for_class(_io_priority, uint32_t(shares));
+    _inflight_update = const_cast<::io_priority_class&>(_io_priority).update_shares(uint32_t(shares));
```

Comment:
I don't like this const_cast. Why do we need this?
Unless I'm missing something, io_priority_class only holds just an _id, and that's it - not the shares. Since modifying the shares doesn't modify this id, the update_shares should work fine on a const reference. Maybe we forgot to allow this in Seastar?

Reply:
I'm not a big fan of const_cast either but that was the best solution I could come up with without changing Seastar API.

I've sent out "[PATCH] io_priority_class: Make update_shares const" to Seastar mailing list and added a FIXME here to remove const_cast once Seastar API is changed.

Maybe we could merge this and remove the const_cast once Seastar API is changed? If it's changed at all because the addition of const to io_priority_class::update_shares is a bit controversial. I think having a clean build with no warnings while  we're discussing Seastar API is valuable - what do you think @nyh ?

Reply:
1. Why is adding the const in Seastar controversial? Who was against it?
2. This patch is optional - it just cleans a warning, doesn't fix any bug or feature - so why hurry with it? If we think that fixing Seastar is better than the ugly "const_cast", then let's wait a few more hours for it. We can commit the two other patches separately from this one (but I'd really appreciate a review from someone who's a bigger expert in this code than me).

Reply:
1. No one is against it at the moment but I can imagine a reasoning that setting shares for io_priority_class actually changes this class because now it gets more/less priority. The alternarive narration that io_priority_class is just an ID is also valid. I'm just saying that there are more than one sane interpretation.
2. I really don't have any problems with few hours :). I'm just not sure how long it takes to push seastar changes and seastar update in Scylla. I somehow assumed it's much longer than just changing Scylla alone. Maybe I had wrong impression.

Reply:
Removed the const_cast